tests/functional/utils/chart_certification.py

```diff
@@ -502,6 +502,22 @@ def process_report(self, update_chart_sha=False, update_url=False, url=None):
                         logging.info(f"Updated chart-uri value in report: {url}")
                     except Exception as e:
                         pytest.fail("Failed to update report yaml with chart-uri")
+            
+            #For removing the check for missing check scenario
+            if missing_check:
+                logging.info(f"Updating report with {missing_check}")
+                with open(report_path, 'r+') as fd:
+                    report_content = yaml.safe_load(fd)
+                    results = report_content["results"]
+                    new_results = []
+                    for result in results:
```

Comment:
Seems like a good fit for `filter`: https://docs.python.org/3/library/functions.html#filter, though this is more readible so I don't have strong opinion.

Reply:
Good suggestion. I think applying filter for this part of the code will be good, added the change.